random_seed.py

```diff
@@ -149,13 +149,34 @@ def generate(self, world_graph = None):
                 elif self.plando_data.magical_seeds_count > logic_settings.magical_seeds_required:
                     logic_settings.magical_seeds_required = self.plando_data.magical_seeds_count
 
+                if logic_settings.starway_chapters_needed_count == -1:
+                    # note: don't roll zero here
+                    logic_settings.starway_chapters_needed_count = random.randint(1, 7)
+                if logic_settings.starway_chapters_needed_count in [0, 7]:
+                    logic_settings.required_chapters = RequiredChapters.ANY
+
                 if logic_settings.starway_spirits_needed_count == -1:
                     # note: don't roll zero here
-                    logic_settings.starway_spirits_needed_count = random.randint(1, 7)
+                    if logic_settings.required_chapters == RequiredChapters.SPECIFIC_AND_LIMITCHAPTERLOGIC:
+                        logic_settings.starway_spirits_needed_count = random.randint(1, logic_settings.starway_chapters_needed_count)
+                    else:
+                        logic_settings.starway_spirits_needed_count = random.randint(1, 7)
+
+                if logic_settings.starbeam_chapters_needed == -1:
+                    # note: don't roll zero here
+                    if logic_settings.required_chapters == RequiredChapters.SPECIFIC_AND_LIMITCHAPTERLOGIC:
+                        logic_settings.starbeam_chapters_needed = 0
```

Comment:
I don't get it, why is starbeam_chapters_needed incompatible with LCL on random chapters needed?

Reply:
It's not really incompatible, it's just always redundant.  
LCL being active means Star Way requires1-6 chapters (If Star Way is set to or randomly rolls 0 or 7 chapters, then LCL turns itself off).  
When LCL is active, then the Star Beam check cannot require more chapters than Star Way. For example, if you set LCL Star Way to 1 chapter, and set Star Beam to 2 chapters, then the seed is always unbeatable because there are no 2 chapters in logic.  
So if LCL is active then the Star Beam location is forced to have the same number of chapters required, or fewer, than Star Way. And since you have to get through Star Way to reach Star Beam, "Star Beam Chapters Needed" is redundant during LCL.  
So I just set it to zero